app/scripts/lib/rpc-method-middleware/handlers/request-accounts.js

```diff
@@ -96,6 +106,20 @@ async function requestEthereumAccountsHandler(
   /* istanbul ignore else: too hard to induce, see below comment */
   if (accounts.length > 0) {
     res.result = accounts;
+    const numberOfConnectedAccounts =
+      getPermissionsForOrigin(origin).eth_accounts.caveats[0].value.length;
+    sendMetrics({
+      event: MetaMetricsEventName.DappViewed,
+      category: MetaMetricsEventCategory.InpageProvider,
+      referrer: {
+        url: origin,
+      },
+      properties: {
+        is_first_visit: true,
```

Comment:
Thanks for reviewing ~ this is because for first time connection, we define it as true. and it will be false when we fire actions after connection, for example: refresh, navigate